galsim/inclinedsersic.py

```diff
@@ -0,0 +1,230 @@
+
+    Initialization
+    --------------
+
+    @param n                  The Sersic index, n.
+    @param inclination        The inclination angle, which must be a galsim.Angle instance
+    @param scale_radius       The scale radius of the disk.  Typically given in arcsec.
+                              This can be compared to the 'scale_radius' parameter of the
+                              galsim.Sersic class, and in the face-on case, the same same scale
+                              radius will result in the same 2D light distribution as with that
+                              class. Exactly one of this and half_light_radius must be provided.
+    @param half_light_radius  The half-light radius of disk when seen face-on. Exactly one of this
+                              and scale_radius must be provided.
+    @param scale_height       The scale height of the exponential disk.  Typically given in arcsec.
+                              [default: None]
+    @param scale_h_over_r     In lieu of the scale height, you may specify the ratio of the
+                              scale height to the scale radius. [default: 0.1]
+    @param flux               The flux (in photons) of the profile. [default: 1]
+    @param trunc              An optional truncation radius at which the profile is made to drop to
+                              zero, in the same units as the size parameter.
+                              [default: 0, indicating no truncation]
+    @param flux_untruncated   Should the provided `flux` and `half_light_radius` refer to the
+                              untruncated profile? See the documentation of the Sersic class for more
+                              details. [default: False]
+    @param gsparams           An optional GSParams argument.  See the docstring for GSParams for
+                              details. [default: None]
+
+    Methods
+    -------
+
+    In addition to the usual GSObject methods, InclinedSersic has the following access methods:
+
+        >>> n = inclined_sersic_obj.getN()
+        >>> inclination = inclined_sersic_obj.getInclination()
+        >>> r0 = inclined_sersic_obj.getScaleRadius()
+        >>> h0 = inclined_sersic_obj.getScaleHeight()
+        >>> hlr = inclined_sersic_obj.getHalfLightRadius()
+    """
+    _req_params = { "inclination" : galsim.Angle, "n" : float }
+    _opt_params = { "scale_height" : float, "scale_h_over_r" : float, "flux" : float }
+    _single_params = [ { "scale_radius" : float , "half_light_radius" : float } ]
+    _takes_rng = False
+
+    def __init__(self, n, inclination, half_light_radius=None, scale_radius=None, scale_height=None,
+                 scale_h_over_r=None, flux=1., trunc=0., flux_untruncated=False, gsparams=None):
+
+        # Check that the scale/half-light radius is valid
+        if scale_radius is not None:
+            if not scale_radius > 0.:
+                raise Exception("scale_radius must be > zero.")
```

Comment:
I think we usually call these kinds of errors due to bad inputs (here and below) ValueErrors rather than more generic Exceptions.